src/main/scala/utils/MatrixUtils.scala

```diff
@@ -18,9 +18,9 @@ object MatrixUtils extends Serializable {
    * @param mat Input matrix.
    * @return Array of rows.
    */
-  def matrixToRowArray[T](mat: DenseMatrix[T]): Array[DenseVector[T]] = {
+  def matrixToRowArray[T : ClassTag](mat: DenseMatrix[T]): Array[DenseVector[T]] = {
     val matT = mat.t
-    (0 until mat.rows).toArray.map(matT(::, _))
+    (0 until mat.rows).toArray.map(x => DenseVector(matT(::, x).toArray))
```

Comment:
Just curious: Does `.copy` or `.toDenseVector` do the same thing ?

Also would be good to add a comment here on why we are doing this